airbyte-integrations/connectors/source-jdbc/src/main/java/io/airbyte/integrations/source/jdbc/AbstractJdbcSource.java

```diff
@@ -316,6 +318,48 @@ public JdbcDatabase createDatabase(final JsonNode config) throws SQLException {
     return database;
   }
 
+  /**
+   * Retrieves connection_properties from config and also validates if custom
+   * jdbc_url parameters overlap with the default properties
+   *
+   * @param config A configuration used to check Jdbc connection
+   * @return A mapping of connection properties
+   */
+  protected Map<String, String> getConnectionProperties(final JsonNode config) {
+    final Map<String, String> customProperties = JdbcUtils.parseJdbcParameters(config, JdbcUtils.JDBC_URL_PARAMS_KEY);
+    final Map<String, String> defaultProperties = getDefaultConnectionProperties(config);
+    assertCustomParametersDontOverwriteDefaultParameters(customProperties, defaultProperties);
+    return MoreMaps.merge(customProperties, defaultProperties);
+  }
+
+  /**
+   * Validates for duplication parameters
+   *
+   * @param customParameters custom connection properties map as specified by each Jdbc source
+   * @param defaultParameters connection properties map as specified by each Jdbc source
+   * @throws IllegalArgumentException
+   */
+  private void assertCustomParametersDontOverwriteDefaultParameters(final Map<String, String> customParameters,
+      final Map<String, String> defaultParameters) {
+    for (final String key : defaultParameters.keySet()) {
+      if (customParameters.containsKey(key) && !Objects.equals(customParameters.get(key), defaultParameters.get(key))) {
+        throw new IllegalArgumentException("Cannot overwrite default JDBC parameter " + key);
+      }
+    }
+  }
+
+  /**
+   * Retrieves default connection_properties from config
+   *
+   * TODO: make this method abstract and add parity features to
+   * destination connectors
+   * @param config A configuration used to check Jdbc connection
+   * @return A mapping of the default connection properties
+   */
+  protected Map<String, String> getDefaultConnectionProperties(final JsonNode config) {
+    return JdbcUtils.parseJdbcParameters(config, "connection_properties", getJdbcParameterDelimiter());
```

Comment:
we use "connection_properties" string in 10+ places in Java code, let's make it a `public static String` constant in JdbcUtils

Reply:
Same comment as with the usage of `ssl` and `sslmode` for constants. I'm happy to make the necessary commit to update these as constants, however I left them out merely to focus on getting the main functionality addressed